aiida/calculations/plugins/transfer.py

```diff
@@ -0,0 +1,179 @@
+        calcinfo.target_path = self.inputs['metadata']['options']['remote_path']
+
+        object_list = self.inputs['instructions'].get_attribute('object_list')
+        calcinfo.retrieve_list = []
+        calcinfo.local_copy_list = []
+        calcinfo.remote_copy_list = []
+
+        # Ideally this would be removed in the case where the copy doesn't need to send
+        # a job to the scheduler. Right now it is needed by the engine so I have to leave
+        # it here. I also have some commented sections that might be useful later to pass
+        # parameters
+        codeinfo = CodeInfo()
+        codeinfo.code_uuid = self.inputs['code'].uuid
+        # codeinfo.stdout_name = "salidita.out"
+        # codeinfo.stdout_name = self.metadata.options.output_filename
+
+        script_copy = self.inputs['instructions'].get_attribute('script_copy', None)
+        if script_copy:
+            # SHOULD CHECK THAT
+            # (1) NODE IS IN LIST
+            # (2) NODE IS FROM REMOTE
+            # (3) A REMOTE PATH HAS BEEN PROVIDED
+            instruction = object_list.pop(script_copy - 1)
+            source_node_id = instruction[0]
+            source_relpath = instruction[1]
+            target_relpath = instruction[2]
+            source_node = orm.load_node(source_node_id)
+            source_path = os.path.join(source_node.get_remote_path(), source_relpath)
+            target_path = os.path.join(calcinfo.target_path, source_relpath)
+            if not os.path.isabs(calcinfo.target_path):
+                code_node = orm.load_node(self.inputs['code'].uuid)
+                computer_path = code_node.computer.get_workdir()
+                target_path = os.path.join(computer_path, target_path)
+
+            codeinfo.cmdline_params = [source_path, target_path]
+
+            if self.inputs['instructions']['retrieve_files']:
+                calcinfo.retrieve_list.append(target_relpath)
+
+        calcinfo.codes_info = [codeinfo]
+
+        for source_node_id, source_relpath, target_relpath in object_list:
+
+            source_node = orm.load_node(source_node_id)
+
+            if isinstance(source_node, orm.FolderData):
+                source_uuid = source_node.uuid
+                source_path = source_relpath
+                target_path = target_relpath
+                calcinfo.local_copy_list.append((source_uuid, source_path, target_path))
+
+            elif isinstance(source_node, orm.RemoteData):
+                source_uuid = source_node.computer.uuid
+                source_path = os.path.join(source_node.get_remote_path(), source_relpath)
+                target_path = target_relpath
+                calcinfo.remote_copy_list.append((source_uuid, source_path, target_path))
+
+            else:
+                raise ValueError(
+                    'type of source node id={} is not allowed ({})'.format(source_node_id, source_node.node_type)
+                )
```

Comment:
This should never happen due to the validation of the process spec. So this should probably be a `RuntimeError` or something